src/components/views/messages/MPollBody.tsx

```diff
@@ -72,22 +198,122 @@ export default class MPollBody extends React.Component<IBodyProps, IState> {
                                 onChange={this.onOptionSelected}
                             >
                                 <div className="mx_MPollBody_optionVoteCount">
-                                    { _t("%(number)s votes", { number: 0 }) }
+                                    { _t("%(count)s votes", { count: answerVotes }) }
                                 </div>
                                 <div className="mx_MPollBody_optionText">
                                     { answer[TEXT_NODE_TYPE] }
                                 </div>
                             </StyledRadioButton>
                             <div className="mx_MPollBody_popularityBackground">
-                                <div className="mx_MPollBody_popularityAmount" />
+                                <div
+                                    className="mx_MPollBody_popularityAmount"
+                                    style={{ "width": `${answerPercent}%` }}
+                                />
                             </div>
                         </div>;
                     })
                 }
             </div>
             <div className="mx_MPollBody_totalVotes">
-                { _t( "Based on %(total)s votes", { total: 0 } ) }
+                { _t( "Based on %(count)s votes", { count: totalVotes } ) }
             </div>
         </div>;
     }
 }
+
+export class UserVote {
+    ts: number;
+    sender: string;
+    answers: Array<string>;
+
+    constructor(ts: number, sender: string, answers: string[]) {
+        this.ts = ts;
+        this.sender = sender;
+        this.answers = answers;
+    }
+}
+
+function userResponseFromPollResponseEvent(event: MatrixEvent): UserVote {
+    const pr = event.getContent() as IPollResponse;
+    const answers = pr[POLL_RESPONSE_EVENT_TYPE.name].answers;
+
+    return new UserVote(
+        event.getTs(),
+        event.getSender(),
+        answers,
+    );
+}
+
+export function allVotes(pollRelations: Relations): Array<UserVote> {
+    function isPollResponse(responseEvent: MatrixEvent) {
+        return (
+            responseEvent.getType() === POLL_RESPONSE_EVENT_TYPE.name &&
+            responseEvent.getContent().hasOwnProperty(POLL_RESPONSE_EVENT_TYPE.name)
+        );
+    }
+
+    if (pollRelations) {
+        return pollRelations.getRelations().filter(isPollResponse).map(
+            userResponseFromPollResponseEvent);
```

Comment:
suggestion for clarity:
```suggestion
        return pollRelations.getRelations()
            .filter(isPollResponse)
            .map(userResponseFromPollResponseEvent);
```